packages/python/plotly/plotly/express/_core.py

```diff
@@ -1015,9 +1108,9 @@ def infer_config(args, constructor, trace_patch):
         + ["names", "values", "parents", "ids"]
         + ["error_x", "error_x_minus"]
         + ["error_y", "error_y_minus", "error_z", "error_z_minus"]
-        + ["lat", "lon", "locations", "animation_group"]
```

Comment:
I'm wondering if it's necessary to add it here and to array_attrables, as we're basically rewriting the dataframe here and not passing `path` down any further into the machinery

Reply:
it's because `build_dataframe` is called before `process_dataframe_hierarchy`. This way, it works when a user passes arrays or columns in the path instead of just column names.

Reply:
ah perfect. so... much... magic!